### ISSUE TITLE
currentURL fails in Ember 1.12.0-beta.1 before calls to visit

### ISSUE THREAD

**Issue by backspace** [emberjs/ember.js#10850]
TLDR: calling `currentURL()` in acceptance tests before any calls to `visit()` worked in Ember 1.11 but throws an error in Ember 1.12.0-beta.1.

I have an acceptance test that interacts with a modal-like search interface that is accessible from some routes in the application. I wrote some helper methods to reduce repetition in the tests, they look something like this:

    var searchFor = function(query) {
      showSearch();

      fillIn('.search-pane input', query);
    };

    var showSearch = function() {
      andThen(function() {
        if (currentURL().indexOf('path with access to the search') === -1) {
          visit('path with access to the search')
        }
      });

      // Show the search if not already showing
      andThen(function() {
        if (find('.search-pane input').length === 0) {
          click('.search div');
        }
      });
    };

This way I can `searchFor('something')` and not have to care whether I’m at the right path or whether the search interface is currently showing.

In Ember 1.12.0-beta.1, I can no longer check `currentURL()` to see whether I need to navigate somewhere before searching because it throws `undefined is not a function` at [this line](https://github.com/emberjs/ember.js/blob/d6f42c2d3d51695456d78c66a164f11eb67f7283/packages/ember-testing/lib/helpers.js#L32), specifically from the call to `property_get(router, 'location')`.

Is this a desired change? Maybe it makes sense to require a call to `visit()` before checking the URL, but it was an unexpected change to work around. If a call to `visit()` is to be required, maybe there could be an error message to that effect?

Some demonstrative JS Bins, adapted from the [other current `currentURL` issue](https://github.com/emberjs/ember.js/issues/10784):

[Working in Ember 1.11.0](http://emberjs.jsbin.com/sojunuxewa/2/edit)
[Broken in Ember 1.12.0-beta.1](http://emberjs.jsbin.com/hawopulebu/2/edit)
[Still broken after #10805](http://emberjs.jsbin.com/zofuwoveji/2/edit) (hopefully I have the right script tags there, it’s also copied from the other issue)

**Comment by rwjblue** [comment]
Fixed by #11201.
